### PR TITLE
[Synthetics] Add ssl support

### PR DESCRIPTION
**Pull request by btoueg**
- Bump go-datadog-api
 - Add support for ssl tests
 - Add acceptance test
 - Update doc
 - Have subtype defaults to http for api tests
 - Make sure existing TF files won't diff

### CODE REVIEW

**Comment by btoueg** [comment]
```
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v  -timeout 120m
?       github.com/terraform-providers/terraform-provider-datadog       [no test files]
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProvider_impl
--- PASS: TestProvider_impl (0.00s)
=== RUN   TestAccDatadogScreenboard_update
--- PASS: TestAccDatadogScreenboard_update (8.88s)
=== RUN   TestAccDatadogSyntheticsAPITest_importBasic
=== PAUSE TestAccDatadogSyntheticsAPITest_importBasic
=== RUN   TestAccDatadogSyntheticsSSLTest_importBasic
=== PAUSE TestAccDatadogSyntheticsSSLTest_importBasic
=== RUN   TestAccDatadogSyntheticsBrowserTest_importBasic
=== PAUSE TestAccDatadogSyntheticsBrowserTest_importBasic
=== RUN   TestAccDatadogSyntheticsAPITest_Basic
--- PASS: TestAccDatadogSyntheticsAPITest_Basic (5.96s)
=== RUN   TestAccDatadogSyntheticsAPITest_Updated
--- PASS: TestAccDatadogSyntheticsAPITest_Updated (10.23s)
=== RUN   TestAccDatadogSyntheticsSSLTest_Basic
--- PASS: TestAccDatadogSyntheticsSSLTest_Basic (5.93s)
=== RUN   TestAccDatadogSyntheticsSSLTest_Updated
--- PASS: TestAccDatadogSyntheticsSSLTest_Updated (10.21s)
=== RUN   TestAccDatadogSyntheticsBrowserTest_Basic
--- PASS: TestAccDatadogSyntheticsBrowserTest_Basic (6.98s)
=== RUN   TestAccDatadogSyntheticsBrowserTest_Updated
--- PASS: TestAccDatadogSyntheticsBrowserTest_Updated (10.49s)
=== CONT  TestAccDatadogSyntheticsAPITest_importBasic
=== CONT  TestAccDatadogSyntheticsBrowserTest_importBasic
=== CONT  TestAccDatadogSyntheticsSSLTest_importBasic
--- PASS: TestAccDatadogSyntheticsSSLTest_importBasic (6.15s)
--- PASS: TestAccDatadogSyntheticsAPITest_importBasic (6.18s)
--- PASS: TestAccDatadogSyntheticsBrowserTest_importBasic (6.67s)
PASS
ok      github.com/terraform-providers/terraform-provider-datadog/datadog       65.375s
```

**Comment by btoueg** [comment]
```==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v  -timeout 120m
?       github.com/terraform-providers/terraform-provider-datadog       [no test files]
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProvider_impl
--- PASS: TestProvider_impl (0.00s)
=== RUN   TestAccDatadogScreenboard_update
--- PASS: TestAccDatadogScreenboard_update (8.08s)
=== RUN   TestAccDatadogSyntheticsAPITest_importBasic
=== PAUSE TestAccDatadogSyntheticsAPITest_importBasic
=== RUN   TestAccDatadogSyntheticsSSLTest_importBasic
=== PAUSE TestAccDatadogSyntheticsSSLTest_importBasic
=== RUN   TestAccDatadogSyntheticsBrowserTest_importBasic
=== PAUSE TestAccDatadogSyntheticsBrowserTest_importBasic
=== RUN   TestAccDatadogSyntheticsAPITest_Basic
--- PASS: TestAccDatadogSyntheticsAPITest_Basic (6.07s)
=== RUN   TestAccDatadogSyntheticsAPITest_Updated
--- PASS: TestAccDatadogSyntheticsAPITest_Updated (11.13s)
=== RUN   TestAccDatadogSyntheticsSSLTest_Basic
--- PASS: TestAccDatadogSyntheticsSSLTest_Basic (6.04s)
=== RUN   TestAccDatadogSyntheticsSSLTest_Updated
--- PASS: TestAccDatadogSyntheticsSSLTest_Updated (11.99s)
=== RUN   TestAccDatadogSyntheticsBrowserTest_Basic
--- PASS: TestAccDatadogSyntheticsBrowserTest_Basic (6.29s)
=== RUN   TestAccDatadogSyntheticsBrowserTest_Updated
--- PASS: TestAccDatadogSyntheticsBrowserTest_Updated (10.68s)
=== CONT  TestAccDatadogSyntheticsAPITest_importBasic
=== CONT  TestAccDatadogSyntheticsBrowserTest_importBasic
=== CONT  TestAccDatadogSyntheticsSSLTest_importBasic
--- PASS: TestAccDatadogSyntheticsSSLTest_importBasic (6.35s)
--- PASS: TestAccDatadogSyntheticsBrowserTest_importBasic (6.64s)
--- PASS: TestAccDatadogSyntheticsAPITest_importBasic (6.72s)
PASS
ok      github.com/terraform-providers/terraform-provider-datadog/datadog       67.016s```

File: datadog/resource_datadog_synthetics_test_.go [hunk 1]
```diff
@@ -30,6 +31,20 @@ func resourceDatadogSyntheticsTest() *schema.Resource {
 				Required:     true,
 				ValidateFunc: validation.StringInSlice(syntheticsTypes, false),
 			},
+			"subtype": {
+				Type:     schema.TypeString,
+				Optional: true,
+				DiffSuppressFunc: func(key, old, new string, d *schema.ResourceData) bool {
+					if key == "subtype" {
```
Comment (nmuesch) [inline datadog/resource_datadog_synthetics_test_.go]: Any reason to check the key? Isn't this tied directly to the schema (i.e. is there a case where this wouldn't be `subtype`?

File: datadog/resource_datadog_synthetics_test_.go [hunk 2]
```diff
@@ -30,6 +31,20 @@ func resourceDatadogSyntheticsTest() *schema.Resource {
...
+						if d.Get("type") == "api" && old == "http" && new == "" {
```
Comment (nmuesch) [inline datadog/resource_datadog_synthetics_test_.go]: Do we have to check the old key here? Shouldn't we always default to http if new is an empty string and the type is "api"?

Also, I don't know the answer to this, but does TF guarantee the order here? Is `type` always available when this code executes?

File: datadog/resource_datadog_synthetics_test_.go [hunk 3]
```diff
@@ -104,6 +119,15 @@ func syntheticsTestRequest() *schema.Schema {
 					Optional: true,
 					Default:  60,
 				},
+				"host": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+				"port": {
```
Comment (nmuesch) [inline datadog/resource_datadog_synthetics_test_.go]: if port isn't always required (based on the type) should it have a default?

**Comment by nmuesch** [comment]
Also, can we move this PR over to the terraform-providers org/repo 😄

File: website/docs/r/synthetics.html.markdown [hunk 4]
```diff
@@ -77,15 +109,23 @@ resource "datadog_synthetics_test" "bar" {
...
+  - `method` - (Optional) For type=api and subtype=http, one of DELETE, GET, HEAD, OPTIONS, PATCH, POST, PUT
   - `url` - (Required) Any url
-  - `timeout` - (Optional) For type=api, any value between 0 and 60 (Default = 60). No-op for type=browser
+  - `timeout` - (Optional) For type=api, any value between 0 and 60 (Default = 60)
   - `body` - (Optional) Request body
+- `request` - (Required) if type=api and subtype=ssl
+  - `host` - (Required) host name
+  - `port` - (Required) port number
+  - `timeout` - (Optional) For type=api, any value between 0 and 60 (Default = 60)
+- `request` - (Required) if type=api and subtype=ssl
```
[the remark this answers is not in the document]
Comment (btoueg) [reply website/docs/r/synthetics.html.markdown]: Fixed

File: website/docs/r/synthetics.html.markdown [hunk 5]
```diff
@@ -43,6 +44,37 @@ resource "datadog_synthetics_test" "foo" {
 }
 ```
 
+## Example Usage (Synthetics SSL test)
+
+```hcl
+# Create a new Datadog Synthetics API test on https://www.example.org
```
[the remark this answers is not in the document]
Comment (btoueg) [reply website/docs/r/synthetics.html.markdown]: Fixed

File: datadog/resource_datadog_synthetics_test_.go (hunk 3 above)
Comment (btoueg) [reply datadog/resource_datadog_synthetics_test_.go]: Based on the type, port might not even exist, so no sane default AFAIK

File: datadog/resource_datadog_synthetics_test_.go (hunk 1 above)
Comment (btoueg) [reply datadog/resource_datadog_synthetics_test_.go]: Seemed consistent but you're probably right 👍

**Comment by btoueg** [comment]
Closed in favor of https://github.com/terraform-providers/terraform-provider-datadog/pull/279

**Pull request closed without merging**